File: oep_spring_backend/src/main/java/com/oep/dtos/UpdateUserDto.java
package com.oep.dtos;

import jakarta.validation.constraints.Email;
import jakarta.validation.constraints.NotBlank;
import lombok.Data;

@Data
public class UpdateUserDto {
	@NotBlank(message = "Username cannot be blank.")
	private String name;
	@NotBlank(message = "Email cannot be blank.")
	@Email(message = "Invalid Email format.")
	private String email;
	@NotBlank
	private String role;
	@NotBlank
	private String status;
	private Long batchId;
}

File: AdminServiceDotNET/Dtos/UpdateUserDto.cs
using System.ComponentModel.DataAnnotations;

namespace AdminServiceDotNET.Dtos
{
    public class UpdateUserDto
    {
        public long? id { get; set; }
        
        [Required(ErrorMessage = "Name is required")]
        [StringLength(100, ErrorMessage = "Name must not exceed 100 characters")]
        public string name { get; set; }
        
        [Required(ErrorMessage = "Email is required")]
        [EmailAddress(ErrorMessage = "Invalid email format")]
        public string email { get; set; }
        
        [Required(ErrorMessage = "Role is required")]
        public string role { get; set; }
        
        [Required(ErrorMessage = "Status is required")]
        public string status { get; set; }
        
        public long? batchId { get; set; }
    }
}